src/System.Buffers.Primitives/System/Buffers/ReaderWriter/BufferReader_search.cs

```diff
@@ -0,0 +1,154 @@
+        {
+            var unread = UnreadSegment;
+            var index = unread.IndexOf(value);
+            if (index != -1)
+            {
+                _currentSegmentConsumedBytes += index;
+                var result = _currentSegmentPosition + _currentSegmentConsumedBytes;
+                _currentSegmentConsumedBytes++; // skip delimiter
+                return result;
+            }
+
+            var nextPosition = _nextSegmentPosition;
+            var currentPosition = _currentSegmentPosition;
+            var previousPosition = _nextSegmentPosition;
+            while (_sequence.TryGet(ref _nextSegmentPosition, out ReadOnlyMemory<byte> memory))
+            {
+                var span = memory.Span;
+                index = span.IndexOf(value);
+                if (index != -1)
+                {
+                    _currentSegmentPosition = previousPosition;
+                    _currentSegment = span;
+                    _currentSegmentConsumedBytes = index + 1;
+                    return _currentSegmentPosition + index;
+                }
+                previousPosition = _nextSegmentPosition;
+            }
+
+            _nextSegmentPosition = nextPosition;
+            _currentSegmentPosition = currentPosition;
+            return null;
+        }
+
+        private Position? PositionOf(ReadOnlySpan<byte> value)
+        {
+            var unread = UnreadSegment;
+            var index = unread.IndexOf(value);
+            if (index != -1) return _currentSegmentPosition + (_currentSegmentConsumedBytes + index);
+            if (value.Length == 0) return default;
+
+            Span<byte> temp = stackalloc byte[(value.Length - 1) * 2];
+            var currentSegmentPosition = _currentSegmentPosition;
+            var nextSegmentPosition = _nextSegmentPosition;
+            var currentSegmentConsumedBytes = _currentSegmentConsumedBytes;
+
+            while (true) {
+                // Try Stitched Match
+                int tempStartIndex = unread.Length - Math.Min(unread.Length, value.Length - 1);
+                var candidatePosition = currentSegmentPosition + (currentSegmentConsumedBytes + tempStartIndex);
+                int copied = CopyFromPosition(_sequence, candidatePosition, temp);
+                if (copied < value.Length) return null;
+                index = temp.Slice(0, copied).IndexOf(value);
+                if (index < value.Length && index != -1) return candidatePosition + index;
+
+                // Try Next Segment
+                currentSegmentPosition = nextSegmentPosition;
+                if (!_sequence.TryGet(ref nextSegmentPosition, out var memory, true)) return default;
+                currentSegmentConsumedBytes = 0;
+                unread = memory.Span;
+                index = unread.IndexOf(value);
+                if (index != -1) return currentSegmentPosition + index;
+            }
+        }
+        #endregion
+    }
+
+    struct PositionRange
```

Comment:
Seems suspiciously similar to ROB

Reply:
?

Reply:
Use ROB, why do we need another version of the basically same type?